includes/Products/FBCategories.php

```diff
@@ -85,34 +99,74 @@ public function is_valid_value_for_attribute( $category_id, $attribute_key, $val
 		}
 	}
 
-	public function get_category( $id ) {
+	/**
+	 * Fetches given category.
+	 *
+	 * @param string $category_id Id of the category we want to fetch.
+	 *
+	 * @return null|array Null if category was not found or the category array.
+	 */
+	public function get_category( $category_id ) {
 		$this->ensure_data_is_loaded();
-		if ( $this->is_category( $id ) ) {
-			return $this->categories_data[ $id ];
+		if ( $this->is_category( $category_id ) ) {
+			return GoogleProductTaxonomy::TAXONOMY[ $category_id ];
 		} else {
 			return null;
 		}
 	}
 
-	public function get_category_depth( $id ) {
-		if ( ! $this->is_category( $id ) ) {
-			return 0;
+	/**
+	 * Checks if category is root category - it has no parents.
+	 *
+	 * @param string $category_id   Id of the category for which attribute we want to check the value.
+	 *
+	 * @return null|boolean Null if category was not found or boolean that determines if this is a root category or not.
+	 */
+	public function is_root_category( $category_id ) {
+		if ( ! $this->is_category( $category_id ) ) {
+			return null;
 		}
-		$category = $this->get_category( $id );
-		return count( explode( '>', $category ) );
+
+		$category = $this->get_category( $category_id );
+		return empty( $category['parent'] );
 	}
 
-	public function get_category_with_attrs( $id ) {
-		$this->ensure_data_is_loaded();
-		if ( $this->is_category( $id ) ) {
-			return $this->attributes_data[ $id ];
+	/**
+	 * Get attributes for the category.
+	 *
+	 * @param string $category_id   Id of the category for which we want to fetch attributes.
+	 *
+	 * @return null|boolean Null if category was not found or boolean that determines if this is a root category or not.
+	 */
+	public function get_category_with_attrs( $category_id ) {
+		$this->ensure_data_is_loaded( true );
+		if ( $this->is_category( $category_id ) ) {
+			return $this->attributes_data[ $category_id ];
 		} else {
 			return null;
 		}
 	}
 
-	public function is_category( $id ) {
+	/**
+	 * Checks if given category id is valid.
+	 *
+	 * @param string $category_id   Id of the category which we check.
+	 *
+	 * @return boolean Is the id a valid category id.
+	 */
+	public function is_category( $category_id ) {
+		$this->ensure_data_is_loaded();
+		return isset( GoogleProductTaxonomy::TAXONOMY[ $category_id ] );
```

Comment:
I just noticed this:

<img width="961" alt="Screen Shot 2021-04-01 at 2 48 36 pm" src="https://user-images.githubusercontent.com/1892137/113242841-6e4d3e00-92f9-11eb-848c-08e781f75f34.png">

Reply:
I don't think that this matters for this plugin. Since the plugin is using the action scheduler this means that we only support WC versions that support PHP7 as their minimum version.